app/src/main/java/ch/epfl/sdp/blindly/main_screen/profile/ProfilePageFragment.kt

```diff
@@ -0,0 +1,119 @@
+import ch.epfl.sdp.blindly.EditProfile
+import ch.epfl.sdp.blindly.R
+import ch.epfl.sdp.blindly.ViewModelAssistedFactory
+import ch.epfl.sdp.blindly.recording.RecordingActivity
+import ch.epfl.sdp.blindly.settings.Settings
+import ch.epfl.sdp.blindly.user.User
+import ch.epfl.sdp.blindly.user.UserHelper
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class ProfilePageFragment : Fragment() {
+
+    @Inject
+    lateinit var userHelper: UserHelper
+    @Inject
+    lateinit var assistedFactory: ViewModelAssistedFactory
+
+    private lateinit var viewModel: ProfilePageViewModel
+
+    companion object {
+        private const val TAG = "ProfilePage"
+        const val BOUNCE_DURATION: Long = 100
+        private const val ARG_COUNT = "profileArgs"
+        private var counter: Int? = null;
+
+        fun newInstance(counter: Int): ProfilePageFragment {
+            val fragment = ProfilePageFragment();
+            val args = Bundle();
+            args.putInt(ARG_COUNT, counter);
+            fragment.arguments = args;
+            return fragment
+        }
+    }
+
+    /**
+     * Set up the viewModel
+     */
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        if (arguments != null) {
+            counter = requireArguments().getInt(ARG_COUNT)
+        }
+
+        val bundle = Bundle()
+        bundle.putString("uid", userHelper.getUserId())
+
+        val viewModelFactory = assistedFactory.create(this, bundle)
+
+        viewModel = ViewModelProvider(this, viewModelFactory)[ProfilePageViewModel::class.java]
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+
+        return inflater.inflate(R.layout.fragment_profile_page, container, false)
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val userInfoText = view.findViewById<TextView>(R.id.user_info_text)
+        val userDescriptionText = view.findViewById<TextView>(R.id.user_description_text)
+
+        val context = this@ProfilePageFragment.context
+
+        val editButton = view.findViewById<Button>(R.id.edit_info_profile_button)
+        setOnClickListener(editButton, Intent(context, EditProfile::class.java))
+
+        val recordAudioButton = view.findViewById<Button>(R.id.record_audio_profile_button)
+        setOnClickListener(recordAudioButton, Intent(context, RecordingActivity::class.java))
+
+        val settingsButton = view.findViewById<Button>(R.id.settings_profile_button)
+        setOnClickListener(settingsButton, Intent(context, Settings::class.java))
+
+        viewModel.user.observe(viewLifecycleOwner) {
+            userInfoText.text = "${it.username}, ${User.getUserAge(it)}"
```

Comment:
It seems that the way you concatenate the text is not a good idea, according to Cirrus.